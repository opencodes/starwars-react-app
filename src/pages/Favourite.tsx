import CharacterCard from '../components/CharacterCard'
import { useAppContext } from '../context/AppContext'
import { ICharacter } from '../interface/swapi.interface'
import Head from './components/Head'

type Props = {}

const Favourite = (props: Props) => {
    const { favouriteCharacters, setFavouriteCharacters } = useAppContext();
    const saveToFavourite = (item: ICharacter) => {
        let updatedList = favouriteCharacters.filter((f: ICharacter) => f.id !== item.id);
        setFavouriteCharacters(updatedList)
    }
    return (
        <>
            <Head onSearch={() => { }} />
            <div className="container mt-5">
                <h5>Favourite Characters</h5>
                <hr />
                <div className="d-flex flex-wrap">
                    {favouriteCharacters && favouriteCharacters.map((c: ICharacter, i: number) =>
                        <div style={{ width: '20%' }}
                            key={c.name}
                        >
                            <CharacterCard character={c} onFavouriteClick={() => saveToFavourite(c)} />
                        </div>
                    )}
                </div>
            </div>
        </>
    )
}

export default Favourite