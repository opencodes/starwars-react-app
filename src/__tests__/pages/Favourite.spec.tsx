
import Favourite from '../../pages/Favourite';
import { renderWithRouter } from '../test-utils';
import { } from '@testing-library/react';


describe('Favourite', () => {
    test('should render Favourite Page correctly', () => {
        const { asFragment } = renderWithRouter(<Favourite />);
        expect(asFragment).toMatchSnapshot();
    });
});

