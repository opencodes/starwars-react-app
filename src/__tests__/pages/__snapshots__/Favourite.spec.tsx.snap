// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourite should render Favourite Page correctly 1`] = `[Function]`;
